### PR TITLE
add support for smartos with tests

### PR DESCRIPTION
**Pull request by bixu**
Hi @farmdawgnation,
Here's a PR adding a lot of Test Kitchen bits as well as SmartOS support to your cookbook. I really like the simplicity of your approach and have tried to stick to it while including some emerging standard practices for cookbooks.

The biggest changes you'll notice is that I'm doing paths in an attribute-driven way and also allowing users to choose between the 'java' and 'openjdk' cookbooks as dependencies via an attribute.

This reason for no longer hardcoding the 'java' dependency is that the big, community 'java' cookbook has issues when run inside containers and also on SmartOS.

Feel free to ask for adjustments or to ask questions.  Thanks again for the initial work.  We hope to start using the code here at yetu starting this week :smiley:

### CODE REVIEW

**Comment by farmdawgnation** [comment]
Thanks for the contribution. I'll try to fit in taking a look at this today. :)

**Comment by bixu** [comment]
You can run the test by doing something like (after installing VirtualBox and Vagrant):
```
cd chef-sbt
gem install test-kitchen
kitchen test
```

**Comment by farmdawgnation** [comment]
Wow. So sorry. This slipped through the cracks.

File: attributes/default.rb [hunk 1]
```diff
@@ -17,5 +17,13 @@
...
-default['sbt']['version'] = '0.12.4'
+default['sbt']['version'] = '0.13.8'
 default['sbt']['java_options'] = '-Xms512M -Xmx1536M -Xss1M -XX:+CMSClassUnloadingEnabled -XX:MaxPermSize=512M'
+default['sbt']['java_cookbook'] = 'openjdk'
+
+case node['platform']
+when 'smartos'
+  default['sbt']['path'] = '/opt/local'
+when 'ubuntu'
+	default['sbt']['path'] = '/usr/local'
```
Comment (farmdawgnation) [inline attributes/default.rb]: Looks like there's a hard tab on this one.

**Comment by farmdawgnation** [comment]
Aside from a hard tab this all looks good. I'm not set up to run the test myself right now, but I will try to get to it soon. Thanks again.

**Pull request merged**